aten/src/ATen/SparseTensorImpl.h

```diff
@@ -45,28 +40,14 @@ struct AT_API SparseTensorImpl : public TensorImpl {
   Tensor indices() const { return indices_; }
   Tensor values() const { return values_; }
 
-  IntList sizes() const override;
-  IntList strides() const override;
-  bool is_contiguous() const override;
-  int64_t size(int64_t d) const override;
-  int64_t stride(int64_t d) const override;
-  void resize_dim(int64_t ndim) override;
-  void set_size(int64_t dim, int64_t new_size) override;
-  void set_stride(int64_t dim, int64_t new_stride) override;
-  void set_storage_offset(int64_t storage_offset) override;
-
   int64_t dim() const override;
-  TensorImpl* maybe_zero_dim(bool condition_when_zero_dim) override;
-  const Storage& storage() const override;
-  int64_t storage_offset() const override;
 
   // WARNING: This function does NOT preserve invariants of sparseDims/denseDims with
   // respect to indices and values
   void raw_resize_(int64_t sparseDims, int64_t denseDims, IntList size) {
-    size_ = size.vec();
+    set_sizes_and_strides(size.vec(), {});
```

Comment:
the previous usages of set_sizes_and_strides always set the strides, but now it's overloaded in a pretty subtle way.  How about changing the type of strides to optional<IntList> and passing at::nullopt here and checking that if the thing didn't have strides, that nullopt was passed?